### DISCUSSION TITLE
Pi Zero 2W hangs on compile

### DISCUSSION THREAD

**Discussion by mntbighker** [bluenviron/mediamtx#3354] [Questions] [answered]
### Question

I have built mediamtx 1.8.1 on Pi Zero W several times. The pre-built executable fails because of issues with libcamera0 apparently. It was running my 2 zero's so hot I ordered two 64bit Zero 2W's. Now the issue I seem to be having is when I run "go build -tags rpicamera ." and have top running it appears to use up all of the memory and all of the swap, then hangs. This did not happen on the 32 bit Raspbian Bookworm. Any suggestions on getting this to compile?  For the record, I do have a Pi400 as well. Do I need to compile on the 400 and copy the binaries over?

**Comment by mntbighker** [comment, marked as answer]
I increased swap from 100 to 500 and the compile completed. The system load on 64 bit Pi Zero 2W is barely worth mentioning compared to Pi Zero W.

**Reply by aler9** [reply]
You can also build the server from a more powerful machine, this is called "cross-compiling" and is documented here:
https://github.com/bluenviron/mediamtx?tab=readme-ov-file#cross-compile

**Reply by aler9** [reply]
or even better, if you want support for the RPI camera, use "make binaries":
https://github.com/bluenviron/mediamtx?tab=readme-ov-file#compile-for-all-supported-platforms

[a comment by an automated account is not shown]
